src/apps/cardano/sign_tx.py

```diff
@@ -74,12 +74,16 @@ def is_change(output, inputs):
 
     try:
         # request transactions
-        transactions = []
         tx_req = CardanoTxRequest()
+        transaction = Transaction(
+            msg.inputs, msg.outputs, keychain, msg.protocol_magic
+        )
+
         for index in range(msg.transactions_count):
             progress.advance()
             tx_ack = await request_transaction(ctx, tx_req, index)
-            transactions.append(tx_ack.transaction)
+            tx_hash = hashlib.blake2b(data=bytes(tx_ack.transaction), outlen=32).digest()
+            transaction.tx_data[tx_hash] = cbor.decode(tx_ack.transaction)
```

Comment:
hmm, if I understand right, this ends up storing all decoded input transactions in memory at once. We could avoid that, the idea is to set a flag "attested" for each of the msg.inputs, that have a hash, index and amount matching with the currently decoded transaction. That way you can forget that decoded transaction right after processing it. This should yield significantly better results than the current approach